podman/domain/containers.py

```diff
@@ -413,20 +414,9 @@ def stats(self, **kwargs) -> Iterator[Union[bytes, Dict[str, Any]]]:
         response.raise_for_status()
 
         if stream:
-            return self._stats_helper(decode, response.iter_lines())
+            return api.stream_helper(response, decode_to_json=decode)
 
-        return json.loads(response.text) if decode else response.content
-
-    @staticmethod
-    def _stats_helper(
-        decode: bool, body: Iterator[bytes]
-    ) -> Iterator[Union[bytes, Dict[str, Any]]]:
-        """Helper needed to allow stats() to return either a generator or a bytes."""
-        for entry in body:
-            if decode:
-                yield json.loads(entry)
-            else:
-                yield entry
+        return json.loads(response.content) if decode else response.content
```

Comment:
This changes the behaviour of stats() to read all the contents before returning the JSON payload. Given there is a streaming flag I don't see this change as wrong, just potentially breaking for developers who may be using the previous implementation.

Reply:
The change in loading the response was made in 3ad81a73
It has already been merged in #237 and released. 😅 

Also the existing code had issues for non-stream mode, where it was encoding each in the response with an `\n` appended. I ended up replacing it with a simple `json.loads`. 

Snippet for Ref:
```py
        with io.StringIO() as buffer:
            for entry in response.text:
                buffer.write(json.dumps(entry) + "\n")
            return buffer.getvalue()
```

Please feel free to suggest any better alternatives. 
I'll make the required changes.